bundles/io/org.eclipse.smarthome.io.encryption/src/main/java/org/eclipse/smarthome/io/crypto/internal/CryptoModule.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.crypto.internal;
+
+import java.io.UnsupportedEncodingException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.KeySpec;
+import java.util.Base64;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.PBEParameterSpec;
+import javax.xml.bind.DatatypeConverter;
+
+import org.eclipse.smarthome.io.crypto.Crypto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ *
+ * @author Pauli Anttila - Initial contribution
+ *
+ */
+public class CryptoModule implements Crypto {
+    public final static String ENV_VARIABLE_ENC_PASSPHRASE = "CFG_ENCRYPTION_PASSPHRASE";
+
+    private final Logger logger = LoggerFactory.getLogger(CryptoModule.class);
+
+    private byte[] salt = { (byte) 0xB8, (byte) 0x62, (byte) 0xF2, (byte) 0x29, (byte) 0x85, (byte) 0xE8, (byte) 0x11,
```

Comment:
Please make this a contant.

Reply:
I recon the salt should not be a constant (visible for everyone that can read source codes, including the bad guys), but be an initially computed value (hashing the mac address, serial no of the cpu or anything). That way the salt is at least different for every user and not public available.

Reply:
A salt is not a secret, it just protects the user against pre-computed rainbow tables for the used hash algorithm. But I agree that having an individual salt per installation is even better than a global one for all ESH installations.